Refactored rank-specific logger

1darshanpatil opened the pull request:
#Refactored rank_specific functions.
#Implemented the logic without changing the behavior. 
#The readability and maintainability of the code improved.
 -------------------------
Tested against:
test_torch_logging.py

1darshanpatil committed:
Refactored rank-specific logger

#Refactored rank_specific functions.
#Implemented the logic without changing the behavior.
#The readability and maintainability of the code improved.
-------------------------
Tested against:
test_torch_logging.py

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/twitter/the-algorithm-ml?pullRequest=120) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/twitter/the-algorithm-ml?pullRequest=120) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/twitter/the-algorithm-ml?pullRequest=120) it.</sub>

1darshanpatil committed:
escaped conditional trap with for loop

optimizer selection logic for improved readability and maintainability.

1darshanpatil committed:
Early exit

used the any function to short-circuit the loop as soon as an empty tensor is found.

1darshanpatil commented:
used the `any` function to short-circuit the loop as soon as an empty tensor is found.

captain-hoity-toity commented on ml_logging/torch_logging.py:
```diff
@@ -17,45 +17,46 @@
 import torch.distributed as dist
```
The more the merrier doesn't really apply to lines of code.
